engineio/src/asynchronous/async_socket.rs

```diff
@@ -23,7 +26,7 @@ use crate::{
 #[derive(Clone)]
 pub struct Socket {
     handle: Handle,
-    transport: Arc<AsyncTransportType>,
+    transport: Arc<Mutex<AsyncTransportType>>,
```

Comment:
suggestion: a mutex lock shouldn't be needed here, I managed (with some difficultly) to remove it before.